mgmt_cli_test.py

```diff
@@ -14,94 +14,117 @@
 # Copyright (c) 2016 ScyllaDB
 
 import os
-import time
 import re
+import time
 from random import randint
+from pathlib import Path
+from functools import cached_property
 
 from invoke import exceptions
 from pkg_resources import parse_version
 
 from sdcm import mgmt
-from sdcm.sct_events.group_common_events import ignore_no_space_errors
 from sdcm.mgmt import ScyllaManagerError, TaskStatus, HostStatus, HostSsl, HostRestStatus
-from sdcm.mgmt.cli import ScyllaManagerTool, SCYLLA_MANAGER_AGENT_YAML_PATH, update_config_file
+from sdcm.mgmt.cli import ScyllaManagerTool
+from sdcm.remote import shell_script_cmd
+from sdcm.tester import ClusterTester
+from sdcm.cluster import Setup
 from sdcm.nemesis import MgmtRepair
+from sdcm.utils.common import reach_enospc_on_node, clean_enospc_on_node
 from sdcm.sct_events.system import InfoEvent
 from sdcm.sct_events.filters import DbEventsFilter
 from sdcm.sct_events.database import DatabaseLogEvent
-from sdcm.utils.common import reach_enospc_on_node, clean_enospc_on_node
-from sdcm.tester import ClusterTester
+from sdcm.sct_events.group_common_events import ignore_no_space_errors
 
 
 class BackupFunctionsMixIn:
-    is_cred_file_configured = False
-    region = None
-    bucket_name = None
-    DESTINATION = '/tmp/backup'
+    DESTINATION = Path('/tmp/backup')
 
-    @staticmethod
-    def download_files_from_s3(node, destination, file_list):
-        download_cmd = 'aws s3 cp {} {}'
-        for file_path in file_list:
-            node.remoter.run(download_cmd.format(file_path, destination))
+    backup_azure_blob_service = None
+    backup_azure_blob_sas = None
 
-    @staticmethod
-    def download_files_from_gs(node, destination, file_list):
-        download_cmd = 'gsutil cp {} {}'
-        for file_path in file_list:
-            file_path = file_path.replace('gcs://', 'gs://')
-            node.remoter.run(download_cmd.format(file_path, destination))
+    @cached_property
+    def locations(self) -> list[str]:
+        backend = self.params.get("backup_bucket_backend")
+
+        # FIXME: Make it works with multiple locations or file a bug for scylla-manager.
```

Comment:
So, should we drop support for older versions of the agent?

Reply:
Actually, I saw this command, but because I didn't know why it implemented this way, I just mimicked the code for S3 and GCS.

Reply:
So, should we replace all that stuff with `scylla-manager-agent download-files -L <bucket> -T <snapshot> -d <dir> --mode upload` ?

@roydahan ?

Reply:
... and add `-K keyspace1.standard1` also.

Reply:
Is it planned for this PR or later task?

Reply:
As I've done with coding for this PR, imo, it should be a separate task. But we need to decide if we want to remove old code completely or not. Since I'm pretty new to this topic this question is for @roydahan and @ShlomiBalalis